### PR TITLE
Move Integer#positive? and Integer#negative? query methods to Numeric

### PR DESCRIPTION
**Pull request by meinac**
By this way Integer, Rational, Float, Fixnum, Bignum classes have the same behaviour

### CODE REVIEW

**Comment by rafaelfranca** [comment]
Thank you for the pull request. Before doing that I want to get feedback from Ruby core since they already rejected this same feature there and seems there are a really good reason to not include at `Numeric`. See https://bugs.ruby-lang.org/issues/5513

**Comment by meinac** [comment]
@rafaelfranca I can't understand what they are talking about because I don't know Japanese :smile: 
If you understand the reason behind they don't add this into the numeric class, can you please tell me. If they have a good reason for this maybe we can add this feature into the other major numeric classes like float, rational etc.

**Comment by rafaelfranca** [comment]
Yeah, I'm waiting https://bugs.ruby-lang.org/issues/11151

**Comment by meinac** [comment]
As far as I understand from the comment the only reason is `Complex` numbers. What do you think about explicitly writing code for `Complex` class to raise undefined method error?

**Comment by egilburg** [comment]
`Math::DomainError` would be an appropriate method to raise in that case.

**Comment by meinac** [comment]
@egilburg Thank you for your suggestion but the ruby documentation says about `Math::DomainError` that `Raised when a mathematical function is evaluated outside of its domain of definition` it seems ok but I think it's not. 
If you try to compare zero with Complex number like below
```ruby
c = Complex(1)
c < 0
```
It will raise NoMethodError so raising NoMethodError for these methods is more suitable.

**Comment by egilburg** [comment]
In this case you should use undef_method rather than raise no method error. Otherwise respond_to?(:positive?) would still return true which isn't something you want.

**Comment by meinac** [comment]
:+1: Yes, this is a good point :smile:

File: activesupport/test/core_ext/numeric_ext_test.rb [hunk 1]
```diff
@@ -389,4 +389,16 @@ def test_to_s__injected_on_proper_types
   def test_in_milliseconds
     assert_equal 10_000, 10.seconds.in_milliseconds
   end
+
+  def test_positive
```
Comment (egilburg) [inline activesupport/test/core_ext/numeric_ext_test.rb]: You'll want to add test coverage for all the expected classes, e.g. `0`, `0.0`, rationals etc

Comment (meinac) [inline activesupport/test/core_ext/numeric_ext_test.rb]: I'm waiting a response from @rafaelfranca then I'll change the behaviour and add tests for all numeric types.

**Comment by rafaelfranca** [comment]
The feature was added in Ruby iteself so I changed it to be loaded conditionally at 6c55cffa5bb804fc0291ec4aa82527ec3c57f2d2

**Pull request merged**